coralproject/talk
# [CORL-2782] Separate tests into different sections to improve GHA performance

## What does this PR do?

- Maintains existing `npm run test` functionality for devs
    - Devs don't have to do anything different to run tests
- Separates the GHA test runners into sections to improve multi-thread performance of each individual test section
- Prevents timeouts and other test failures when running tests using GitHub Actions

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- run `npm run test`
- run `npm run test:server`
- run `npm run test:client:stream`
- run `npm run test:client:admin`
- run `npm run test:client:other`
- see that all of these pass

- also see that GHA ran these actions and passed

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations
